src/broker/module.c

```diff
@@ -354,14 +355,18 @@ module_t *module_create (flux_t *h,
 
     /* Broker end of PAIR socket is opened here.
      */
-    if (!(p->sock = zsock_new_pair (NULL))) {
+    if (!(p->sock = zmq_socket (p->zctx, ZMQ_PAIR))) {
         errprintf (error, "could not create zsock for %s", p->name);
         goto cleanup;
     }
     zsock_set_unbounded (p->sock);
     zsock_set_linger (p->sock, 5);
-    if (zsock_bind (p->sock, "inproc://%s", module_get_uuid (p)) < 0) {
-        errprintf (error, "zsock_bind inproc://%s", module_get_uuid (p));
+    snprintf (p->endpoint,
+              sizeof (p->endpoint),
+              "inproc://%s",
+              module_get_uuid (p));
```

Comment:
A suggestion to cast to `void` and add a note that maximum size of string can never overflow 127 chars... But up to you.